msprime/mutations.py

```diff
@@ -80,9 +80,9 @@ def mutate(tree_sequence, rate=None, random_seed=None, model=None, keep=False):
 
     By default, sites and mutations in the parameter tree sequence are discarded.
     If the ``keep`` parameter is true, however, *additional* mutations are
-    simulated. To ensure that the returned tree sequence is valid, and
-    mutations generated by this function that have precisely the same position
-    are discarded.
+    simulated. Under the infinite sites mutation model, any sites that are
+    generated with precisely the same position as an existing position will be
+    rejection sampled.
```

Comment:
How about "Under the infinite sites mutation model, all new mutations generated will occur at distinct positions from each other and from any existing mutations (by rejection sampling)."